### PR TITLE
Add Google Test framework

### PR DESCRIPTION
**Pull request by fredlas**
This change gets the Google Test framework nicely integrated into the codebase. To check that it is in fact working, this change also adds test/alignment_datatype_test.cpp. `make test && ./bin/alignment_datatype_test` should show passing Google Test output.

Having written the tests, alignment_datatype.cpp now catches several types of malformed inputs, with helpful error messages.

To get the test framework working with the code, I had to refactor a little to get TagSort's `main()` away from the rest of the code, so that the framework can use its own main. But that was really just moving code out into new files, and renaming existing files; no actual code structure change.

### CODE REVIEW

File: fastqpreprocessing/src/alignment_datatype.cpp [hunk 1]
```diff
@@ -123,24 +140,75 @@ void LineFields::writeTabbedToFile(std::ofstream& outfile)
           << molecule_barcode_base_above_30 << "\n";
 }
 
-std::string LineFields::getNextField(std::string const& s)
+void LineFieldsParser::crashLF(std::string msg)
+{
+  crash(msg + "\nThe bad line:\n" + s_);
+}
+
+int LineFieldsParser::getNextFieldInt()
```
Comment (fredlas) [inline fastqpreprocessing/src/alignment_datatype.cpp:148]: This block here is all actual code change, to fix bugs / better handle corner cases I found through testing. I could split these changes into another PR, if you'd prefer.

**Review by kevinpalis — APPROVED** [review]
Looks great!

**Comment by fredlas** [comment]
Thanks!

**Review by jessicaway — APPROVED** [review]
Looks great! Thanks @fredlas !

**Comment by fredlas** [comment]
Thanks! This ought to be "the" big one, hopefully.
